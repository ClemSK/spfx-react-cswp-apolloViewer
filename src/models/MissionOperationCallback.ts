import { IMission } from '../IMission';

export type MissionOperationCallback = (mission: IMission) => void;
