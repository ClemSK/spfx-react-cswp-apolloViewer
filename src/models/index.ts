export * from './IMission';
export * from './MissionOperationCallback';
