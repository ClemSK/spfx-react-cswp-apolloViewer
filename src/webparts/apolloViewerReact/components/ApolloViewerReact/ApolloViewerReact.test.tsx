import * as React from 'react';
import 'jest';
import { shallow } from 'enzyme';

import { IMission } from '../../../../models';
import { ApolloViewerReact } from './';

let _missionSetA: IMission[] = undefined;
let _missionSetB: IMission[] = undefined;

beforeAll(() => {
  _missionSetA = [
    {
      id: 'AS-510',
      name: 'Apollo 15',
      launch_date: '07/26/1971',
      end_date: '08/07/1971',
      image: 'apollo-15.png',
      wiki_href: 'https://en.wikipedia.org/wiki/Apollo_15',
      summary:
        'First Extended LM and rover, landed in Hadley-Apennine. Surface EVA time:18:33 hr. Samples returned: 169.10 pounds (76.70 kg).',
      crew: [
        {
          role: 'Commander',
          astronaut_id: '26',
        },
        {
          role: 'Pilot',
          astronaut_id: '17',
        },
        {
          role: 'Sr Pilot',
          astronaut_id: '31',
        },
        {
          role: 'Command Module Pilot',
          astronaut_id: '31',
        },
        {
          role: 'Lunar Module Pilot',
          astronaut_id: '17',
        },
      ],
    },
    {
      id: 'AS-511',
      name: 'Apollo 16',
      launch_date: '04/16/1972',
      end_date: '04/27/1972',
      image: 'apollo-16.png',
      wiki_href: 'https://en.wikipedia.org/wiki/Apollo_16',
      summary:
        'Landed in Plain of Descartes. Surface EVA time: 20:14 hr. Samples returned: 207.89 pounds (94.30 kg).',
      crew: [
        {
          role: 'Commander',
          astronaut_id: '32',
        },
        {
          role: 'Pilot',
          astronaut_id: '11',
        },
        {
          role: 'Sr Pilot',
          astronaut_id: '19',
        },
        {
          role: 'Command Module Pilot',
          astronaut_id: '29',
        },
        {
          role: 'Lunar Module Pilot',
          astronaut_id: '11',
        },
      ],
    },
    {
      id: 'AS-512',
      name: 'Apollo 17',
      launch_date: '12/07/1972',
      end_date: '12/19/1972',
      image: 'apollo-17.png',
      wiki_href: 'https://en.wikipedia.org/wiki/Apollo_17',
      summary:
        'Only Saturn V night launch. Landed in Taurus-Littrow. First geologist on the Moon. Final manned Moon landing. Surface EVA time: 22:02 hr. Samples returned: 243.40 pounds (110.40 kg).',
      crew: [
        {
          role: 'Commander',
          astronaut_id: '6',
        },
        {
          role: 'Pilot',
          astronaut_id: '24',
        },
        {
          role: 'Sr Pilot',
          astronaut_id: '13',
        },
        {
          role: 'Command Module Pilot',
          astronaut_id: '13',
        },
        {
          role: 'Lunar Module Pilot',
          astronaut_id: '24',
        },
      ],
    },
  ];
  // slice down to 2 missions
  _missionSetB = _missionSetA.slice(1);
});

test('renders set A of missions', () => {
  const wrapper = shallow(<ApolloViewerReact />);

  wrapper.setState({ missions: _missionSetA });
  expect(wrapper).toMatchSnapshot();

  wrapper.setState({ missions: _missionSetB });
  expect(wrapper).toMatchSnapshot();
});
