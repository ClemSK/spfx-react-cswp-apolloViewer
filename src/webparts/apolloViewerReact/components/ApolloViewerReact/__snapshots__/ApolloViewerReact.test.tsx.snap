// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders set A of missions 1`] = `
<section
  className="apolloViewerReact"
>
  <div
    className="welcome"
  >
    <span>
      Welcome to the React Apollo Missions Web Part!
    </span>
    <ApolloMissionList
      missions={
        Array [
          Object {
            "crew": Array [
              Object {
                "astronaut_id": "26",
                "role": "Commander",
              },
              Object {
                "astronaut_id": "17",
                "role": "Pilot",
              },
              Object {
                "astronaut_id": "31",
                "role": "Sr Pilot",
              },
              Object {
                "astronaut_id": "31",
                "role": "Command Module Pilot",
              },
              Object {
                "astronaut_id": "17",
                "role": "Lunar Module Pilot",
              },
            ],
            "end_date": "08/07/1971",
            "id": "AS-510",
            "image": "apollo-15.png",
            "launch_date": "07/26/1971",
            "name": "Apollo 15",
            "summary": "First Extended LM and rover, landed in Hadley-Apennine. Surface EVA time:18:33 hr. Samples returned: 169.10 pounds (76.70 kg).",
            "wiki_href": "https://en.wikipedia.org/wiki/Apollo_15",
          },
          Object {
            "crew": Array [
              Object {
                "astronaut_id": "32",
                "role": "Commander",
              },
              Object {
                "astronaut_id": "11",
                "role": "Pilot",
              },
              Object {
                "astronaut_id": "19",
                "role": "Sr Pilot",
              },
              Object {
                "astronaut_id": "29",
                "role": "Command Module Pilot",
              },
              Object {
                "astronaut_id": "11",
                "role": "Lunar Module Pilot",
              },
            ],
            "end_date": "04/27/1972",
            "id": "AS-511",
            "image": "apollo-16.png",
            "launch_date": "04/16/1972",
            "name": "Apollo 16",
            "summary": "Landed in Plain of Descartes. Surface EVA time: 20:14 hr. Samples returned: 207.89 pounds (94.30 kg).",
            "wiki_href": "https://en.wikipedia.org/wiki/Apollo_16",
          },
          Object {
            "crew": Array [
              Object {
                "astronaut_id": "6",
                "role": "Commander",
              },
              Object {
                "astronaut_id": "24",
                "role": "Pilot",
              },
              Object {
                "astronaut_id": "13",
                "role": "Sr Pilot",
              },
              Object {
                "astronaut_id": "13",
                "role": "Command Module Pilot",
              },
              Object {
                "astronaut_id": "24",
                "role": "Lunar Module Pilot",
              },
            ],
            "end_date": "12/19/1972",
            "id": "AS-512",
            "image": "apollo-17.png",
            "launch_date": "12/07/1972",
            "name": "Apollo 17",
            "summary": "Only Saturn V night launch. Landed in Taurus-Littrow. First geologist on the Moon. Final manned Moon landing. Surface EVA time: 22:02 hr. Samples returned: 243.40 pounds (110.40 kg).",
            "wiki_href": "https://en.wikipedia.org/wiki/Apollo_17",
          },
        ]
      }
      onDeleteMission={[Function]}
    />
  </div>
</section>
`;

exports[`renders set A of missions 2`] = `
<section
  className="apolloViewerReact"
>
  <div
    className="welcome"
  >
    <span>
      Welcome to the React Apollo Missions Web Part!
    </span>
    <ApolloMissionList
      missions={
        Array [
          Object {
            "crew": Array [
              Object {
                "astronaut_id": "32",
                "role": "Commander",
              },
              Object {
                "astronaut_id": "11",
                "role": "Pilot",
              },
              Object {
                "astronaut_id": "19",
                "role": "Sr Pilot",
              },
              Object {
                "astronaut_id": "29",
                "role": "Command Module Pilot",
              },
              Object {
                "astronaut_id": "11",
                "role": "Lunar Module Pilot",
              },
            ],
            "end_date": "04/27/1972",
            "id": "AS-511",
            "image": "apollo-16.png",
            "launch_date": "04/16/1972",
            "name": "Apollo 16",
            "summary": "Landed in Plain of Descartes. Surface EVA time: 20:14 hr. Samples returned: 207.89 pounds (94.30 kg).",
            "wiki_href": "https://en.wikipedia.org/wiki/Apollo_16",
          },
          Object {
            "crew": Array [
              Object {
                "astronaut_id": "6",
                "role": "Commander",
              },
              Object {
                "astronaut_id": "24",
                "role": "Pilot",
              },
              Object {
                "astronaut_id": "13",
                "role": "Sr Pilot",
              },
              Object {
                "astronaut_id": "13",
                "role": "Command Module Pilot",
              },
              Object {
                "astronaut_id": "24",
                "role": "Lunar Module Pilot",
              },
            ],
            "end_date": "12/19/1972",
            "id": "AS-512",
            "image": "apollo-17.png",
            "launch_date": "12/07/1972",
            "name": "Apollo 17",
            "summary": "Only Saturn V night launch. Landed in Taurus-Littrow. First geologist on the Moon. Final manned Moon landing. Surface EVA time: 22:02 hr. Samples returned: 243.40 pounds (110.40 kg).",
            "wiki_href": "https://en.wikipedia.org/wiki/Apollo_17",
          },
        ]
      }
      onDeleteMission={[Function]}
    />
  </div>
</section>
`;
